clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java

```diff
@@ -425,4 +427,26 @@ static void checkException(IOException e, TestCloseable... closeablesWithExcepti
         }
     }
 
+    @Test(timeout = 120000)
+    public void testRecursiveDelete() throws IOException {
+        Utils.delete(null); // delete of null does nothing.
+
+        // Test that deleting a temporary file works.
+        File tempFile = TestUtils.tempFile();
+        Utils.delete(tempFile);
+        assertFalse(Files.exists(tempFile.toPath()));
+
+        // Test recursive deletes
+        File tempDir = TestUtils.tempDirectory();
+        File tempDir2 = TestUtils.tempDirectory(tempDir.toPath(), "a");
+        TestUtils.tempDirectory(tempDir.toPath(), "b");
+        TestUtils.tempDirectory(tempDir2.toPath(), "c");
+        Utils.delete(tempDir);
+        assertFalse(Files.isDirectory(tempDir.toPath()));
+        assertFalse(Files.isDirectory(tempDir2.toPath()));
+
+        // Test that deleting a non-existent directory hierarchy works.
+        Utils.delete(tempDir);
+        assertFalse(Files.isDirectory(tempDir.toPath()));
```

Comment:
I changed all the `isDirectory` checks to `exists`. Both work, but I think the latter is a bit more explicit about what we're checking.